src/contract.rs

```diff
@@ -397,26 +386,41 @@ fn replace_scope_owner(
     // if more than one owner is specified, removing all of them can potentially cause data loss
     if owners.len() != 1 {
         return Err(ContractError::InvalidScopeOwner {
-            scope_address: scope.scope_id,
+            scope_address: scope.scope_id.clone(),
             explanation: format!(
                 "the scope should only include a single owner, but found: {}",
                 owners.len(),
             ),
         });
     }
-    // The owner of the scope must be the sender to the contract - otherwise, scopes could be released on the behalf of others
-    if let Some(expected_owner) = validate_expected_owner {
+    if let Some(expected) = expected_owner {
         let owner = owners.first().unwrap();
-        if owner.address != expected_owner {
+        if &owner.address != expected {
             return Err(ContractError::InvalidScopeOwner {
-                scope_address: scope.scope_id,
+                scope_address: scope.scope_id.clone(),
```

Comment:
Had to do clones on these now that they're passed by reference, but the clones only occur on errors, so they don't add up